### DISCUSSION TITLE
How to set up pinia orm

### DISCUSSION THREAD

**Discussion by nagisaando** [CodeDredd/pinia-orm#10] [Q&A] [answered]
Hello, I want to use Pinia with Pinia orm in vue 2 but I can not really understand how to set it up. 

I followed the getting started section from vuex orm: https://vuex-orm.org/guide/prologue/getting-started.html#setup, and trying to insert data but looks like it is not working well. 

the step I took, 

1. Installed pinia and pinia orm
2. added code in main.js as:
````
import Vue from "vue";
import App from "./App.vue";
import { createPinia, PiniaVuePlugin } from "pinia";
import PiniaORM from "pinia-orm";

Vue.use(PiniaVuePlugin);

const pinia = createPinia().use(PiniaORM.install());
new Vue({
  render: (h) => h(App),
  pinia,
}).$mount("#app");

````

3. added models for user and post
```
// src/models/user.js
import { Model } from "pinia-orm";

export default class User extends Model {
  static entity = "users";

  static fields() {
    return {
      id: this.uid(),
      name: this.attr(""),
      email: this.attr(""),
    };
  }
}


````

````
// src/models/post.js

import { Model } from "pinia-orm";
import User from "./user";

export default class Post extends Model {
  static entity = "posts";

  static fields() {
    return {
      id: this.uid(),
      user_id: this.attr(null),
      title: this.attr(""),
      body: this.attr(""),
      published: this.attr(false),
      author: this.belongsTo(User, "user_id"),
    };
  }
}

````

4. App.vue
````
// syntax is composition API because I installed composition API and setup script plugin

<script setup>
import { useRepo } from "pinia-orm";
import Post from "@/models/post";

const posts = [
  {
    id: 1,
    title: "Hello, world!",
    body: "Some awesome body...",
    author: {
      id: 1,
      name: "John Doe",
      email: "john@example.com",
    },
  }
];

const postRepo = useRepo(Post);
postRepo.insert({ data: posts });

</script>
````

and this is what I get in pinia 
<img width="594" alt="Screen Shot 2022-04-20 at 2 31 24 PM" src="https://user-images.githubusercontent.com/56172522/164326399-604a841e-75b2-4270-b3b9-967d4e2da9ed.png">

When I follow same thing in vuex orm, I get this 
<img width="738" alt="Screen Shot 2022-04-20 at 2 32 41 PM" src="https://user-images.githubusercontent.com/56172522/164326543-7855be42-4ba2-49db-9aa1-090c92b7cd89.png">
<img width="717" alt="Screen Shot 2022-04-20 at 2 33 00 PM" src="https://user-images.githubusercontent.com/56172522/164326572-3e21e593-51bd-49c3-b48b-2e15353868a3.png">

I also try to register to database as vuex orm but I got error says; 
<img width="376" alt="Screen Shot 2022-04-20 at 2 36 51 PM" src="https://user-images.githubusercontent.com/56172522/164327038-38423334-10c9-46e6-82cb-9cae08e88d44.png">

````
// main.js
import Vue from "vue";
import App from "./App.vue";
import { createPinia, PiniaVuePlugin } from "pinia";
import PiniaORM from "pinia-orm";

import User from "@/models/user";
import Post from "@/models/post";

Vue.use(PiniaVuePlugin);
const database = new PiniaORM.Database();

database.register(useRepo(User));
database.register(useRepo(Post));


const pinia = createPinia().use(PiniaORM.install());
new Vue({
  render: (h) => h(App),
  pinia,
}).$mount("#app");

````

Thank you so much for your great work, and any suggestion is appreciated 🙏

**Comment by CodeDredd** [comment, marked as answer]
Hey @nagisaando ,

you don't need to register anything to the database. Pinia works by design different. I haven't yet implemented a pinia store which can be handled like a vuex-orm database.
In most use cases you dont need such thing as database because pinia is used by composables.

Please also look at the vuex-orm-next doc ->https://next.vuex-orm.org/guide/prologue/getting-started.html#setup because pinia-orm is based on it.

**Reply by nagisaando** [reply]
Ah thank you, I was looking at the wrong doc. Now it's working as expected. Thank you so much!
